### ISSUE TITLE
Create "how to" to define full lifecycle of runtime fields based field creation UX

### ISSUE THREAD

**Issue by mukeshelastic** [elastic/elasticsearch#88756]
### Defining the full lifecycle of runtime fields based field creation UX

Runtime fields provide a powerful way to define data schema after it is ingested, including a full lifecycle of experimentation by defining the field within kibana data view and not persisting it to real data underneath, followed by promoting fields to a real index such that these fields can be used by any user, in any kibana analytics solution operating on that index and finally, being able to index these fields to provide them the superior query performance. 

### "How to" tutorial for schema evolution using runtime fields
Right now, this documentation is spread in bits and pieces across [kibana](https://www.elastic.co/guide/en/kibana/current/discover.html#add-field-in-discover) and [elasticsearch](https://www.elastic.co/guide/en/elasticsearch/reference/8.3/runtime.html) docs and it would be good to create a "how to" tutorial to show case the full lifecycle, starting from field creation in kibana to indexing them for faster performance so that users can use runtime fields for use cases of creating and modifying schema of their data sources.

**Comment by javanna** [comment]
@mukeshelastic and I talked about this and agreed that there is a need for a good story around schema evolution. It is not just a matter of documenting it but rather build it. One step we identified is working on a high-level API that allows to make a runtime field indexed at the data streams level (see #72142). There is quite a bit to be figured out but it's something that we will work towards. Closing this issue as there isn't much we can do for now on the docs side.
